design-documents/graph-ql/coverage/b2b/negotiableQuotes.graphqls

```diff
@@ -174,12 +196,9 @@ input CloseNegotiableQuotesInput {
 }
 
 type CloseNegotiableQuotesOutput {
-    closed_quotes(
-        filter: NegotiableQuoteFilterInput,
-        pageSize: Int = 20,
-        currentPage: Int = 1
-        sort: NegotiableQuoteSortInput
-    ): NegotiableQuotesOutput @doc(description: "Quotes that were just closed")
+    result_status: TriStateMutationResultStatus!
```

Comment:
Is TriState futureproof? Could an UNKNOWN or other status be added down the road? Maybe rename to something like NegotiableQuoteMutationResultStatus?

Reply:
I second this.

Reply:
The idea was to have a shared enum that could be reused elsewhere outside this module for results that could have partial successes, rather than trying to shoehorn everything into a boolean success status.  I could see maybe adding an "unknown" status, but I can't think of any cases where it would be used; if something happens that a result isn't known, isn't that a failure?

This is meant to be used as a top-level flag for whether or not results need to be investigated.  I don't see a case where we'd want more specific details to be included in this enum; anything that needs more precise information should use a module-specific field.  For NegotiableQuote, that means the `operation_results` field.

Reply:
I do like the idea of a shared enum that can be used outside of this module. However, I'm still not a fan of `TriState`. Why not just `MutationResultStatus`?

Reply:
Yeah I'm not a huge fan of `TriState` either, I was just trying to come up with something that was an accurate descriptor of its intended use case.  I wish we could replace all usages of Boolean status results with a shared enum like this (which would make me more comfortable with the `MutationResultStatus` name), but alas we cannot.  I do think that's at least better than TriState.

Reply:
Even if we can't replace all usages, I still think this is an appropriate name for it. Curious what others, namely @cpartica thinks.

Reply:
I prefer `MutationResultStatus` also